### ISSUE TITLE
False positive "Unused dependencies" under certain conditions

### ISSUE THREAD

**Issue by dartess** [webpro/knip#167]
I get a false positive on one of the checks, but only if I turn off a group of other checks...

Code:

```
import type { RouteMatchCallback } from 'workbox-core';
```

Confiig:

```
const config: KnipConfig = {
  entry: [
    'src/core/serviceWorker/service-worker.ts', // service-worker
  ],
  rules: {
    types: 'off',
    nsTypes: 'off',
    enumMembers: 'off',
  },
};
```

Error:

```
Unused dependencies (1)
workbox-core  package.json
```

If I remove any of the shutdowns, the error magically disappears.

Repl: https://github.com/dartess/knip-bugs-repl-2

**Comment by webpro** [comment]
Hah, this was an odd confusing one indeed. When these three all are `off` this triggered some kind of "no types at all" mode, which skipped that import and made the dependency unused.

Changed the trigger to `isProduction` so `workbox-core` will be reported when using `--production`.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.16.2. See [Release 2.16.2](https://github.com/webpro/knip/releases/tag/2.16.2) for release notes._
